### ISSUE TITLE
[New Op] Add TCI (tile.ci / tensor.ci) — contiguous integer sequence generation

### ISSUE THREAD

**Issue by zhangqi-chen** [hw-native-sys/pypto#1095]
### Operation Level

Other (please specify): **both tile-level and tensor-level**. The op must be registered at both layers (`tile.ci` and `tensor.ci`) with a tensor-to-tile conversion, mirroring how other tensor ops lower to their tile counterparts.

### Proposed Name & Signature

IR-level C++ ops: `tile.ci` and `tensor.ci`.

Python IR / DSL wrappers — propose `ci(...)` (matching ISA) with an `arange(...)` convenience alias:

```python
# IR layer — python/pypto/ir/op/tile_ops.py, python/pypto/ir/op/tensor_ops.py
def ci(
    start: int | Expr,
    *,
    shape: Sequence[int | Expr] | None = None,      # required for tensor.ci; tile.ci may receive it via result_type
    dtype: DataType = DataType.INT32,
    descending: bool = False,
    span: Span | None = None,
) -> Call: ...

# DSL layer — python/pypto/language/op/tile_ops.py, python/pypto/language/op/tensor_ops.py
def ci(start: int | Scalar, shape: Sequence[IntLike], dtype: DataType = pl.INT32,
       descending: bool = False) -> Tile: ...         # tile.*
def ci(start: int | Scalar, shape: Sequence[IntLike], dtype: DataType = pl.INT32,
       descending: bool = False) -> Tensor: ...       # tensor.*

# Convenience alias
arange = ci   # so users can write pl.arange(0, shape=[4, 32])
```

### Semantics Description

`TCI` generates a contiguous integer sequence into a destination tile/tensor. For a linearized index `k` over the valid elements:

- Ascending: `dst[k] = start + k`
- Descending: `dst[k] = start - k`

The linearization order depends on tile layout (implementation-defined).

Reference: `pto-isa/docs/isa/TCI.md` and figure `pto-isa/docs/figures/isa/TCI.svg`.

**Inputs**

- `start`: scalar, same dtype as destination
- `shape`: output shape
- `dtype`: destination dtype
- `descending`: bool kwarg (default `false`)

**Output**

- Tile or tensor with shape `shape` and dtype `dtype`, populated per the formula above.

**Constraints (per ISA)**

- `dtype` must be one of `INT32`, `UINT32`, `INT16`, `UINT16`.
- Destination `Cols != 1` (column vectors not supported).
- Hardware uses `dst.GetValidCol()` as sequence length; does not consult `GetValidRow()`.
- Scalar `start` dtype must match destination dtype.

### Example Usage

```python
import pypto.language as pl

@pl.program
class Example:
    @pl.function(type=pl.FunctionType.InCore)
    def main_incore_0(self, out: pl.Out[pl.Tensor[[4, 32], pl.INT32]]) -> pl.Tensor[[4, 32], pl.INT32]:
        # Generate [0, 1, 2, ..., 31] per row (linearization order is layout-defined)
        idx: pl.Tensor[[4, 32], pl.INT32] = pl.tensor.ci(0, shape=[4, 32], dtype=pl.INT32)
        # Or equivalently via the arange alias
        idx2: pl.Tensor[[4, 32], pl.INT32] = pl.arange(0, shape=[4, 32], dtype=pl.INT32)
        return idx
```

A typical use is preparing the `idx` operand for `tile.sort32` / `tensor.sort32`, where the user currently has to materialize the 0..N-1 sequence manually.

### Hardware Support Considerations

- Maps directly to the existing PTO-AS instruction `pto.tci`. The backend codegen is already partially wired: `src/backend/common/pto_ops_common.cpp:456` (`MakeCiCodegenPTO`) and the registration at `src/backend/common/pto_ops_common.cpp:1566` target `tile.ci`, but the IR-level op is not registered (`ir.is_op_registered(\"tile.ci\") == False`). This issue is to add the missing IR/DSL/conversion layers so that codegen actually fires.
- No CPU fallback needed for this hardware-specific sequence generator.
- Runs in `MemorySpace::Vec` on Ascend 910B / A2A3 (aligned with other tile compute ops).

### Motivation / Use Case

1. **`sort32` usability.** `tile.sort32` / `tensor.sort32` require an `idx` tile/tensor initialized to a sequence (`[0, 1, ..., 31]` for FP16 src, `[0, 2, ..., 62]` for FP32 src). Without `arange`/`ci`, users must fabricate the sequence via orchestration, which is awkward and precludes a fully on-device kernel. See `tests/st/runtime/test_sort.py`, which currently constructs `idx` externally.
2. **Index generation for masking / gather.** Many kernels need a row/column index tile — `tensor.gather`, position-encoded ops, softmax length masks, etc. `arange` is a standard primitive in numpy/torch/triton/JAX; its absence forces users to fake it with external tensor inputs.
3. **Backend already supports it.** Codegen for `tile.ci → pto.tci` exists; the front-end and IR gap is the only thing blocking usage. Closing the loop is low-risk and unlocks immediate value.

### Additional Context

- ISA reference: `pto-isa/docs/isa/TCI.md` (ascending/descending contiguous integer sequence, `descending` attribute).
- Existing backend codegen: `src/backend/common/pto_ops_common.cpp:456` and `:1566`.
- Related tests that would benefit: `tests/st/runtime/test_sort.py` (`test_sort32_fp32`, `test_sort32_gather_fp32`).
- Framework analogues: `numpy.arange`, `torch.arange`, `tl.arange` (Triton), `jax.numpy.arange`.

**Suggested implementation scope (Phase A + Phase B of the `add-op` skill):**

- C++ `REGISTER_OP(\"tile.ci\")` with 1 arg (start scalar), `descending` bool attr, `Vec` output memory, dtype restricted to {INT16, UINT16, INT32, UINT32}.
- C++ `REGISTER_OP(\"tensor.ci\")` similarly (no memory spaces).
- Python IR wrappers in `python/pypto/ir/op/{tile,tensor}_ops.py`.
- Python DSL wrappers + `arange` alias in `python/pypto/language/op/{tile,tensor}_ops.py`.
- `RegisterSimple(\"tensor.ci\", \"tile.ci\")` in `src/ir/transforms/op_conversion_registry.cpp`.
- Unit tests + conversion tests.
- Docs update in `docs/en/dev/ir/05-operators.md` + `docs/zh-cn/...`.
